utils/saasherder.py

```diff
@@ -87,6 +89,20 @@ def _collect_namespaces(self):
                     namespaces.append(namespace)
         return namespaces
 
+    def _estimate_available_thread_pool_size(self):
+        # if there are 20 threads and only 3 targets,
+        # each thread can use ~20/3 threads internally.
+        # if there are 20 threads and 100 targts,
+        # each thread can use 1 thread internally.
+        # 
+        # each namespace is in fact a target,
+        # so we can use it to calculate.
+        thread_pool_size = int(
+            self.thread_pool_size /
+            len(self.namespaces)
+        )
+        return max(thread_pool_size, 1)
+
```

Comment:
very efficient usage of the availables threads

Reply:
:heart: